torch/serialization.py

```diff
@@ -397,26 +339,26 @@ def _validate_privateuse1_device(location, backend_name):
     return device_index
 
 
-def _privateuse1_deserialize(obj, location):
-    backend_name = torch._C._get_privateuse1_backend_name()
-    if location.startswith(backend_name):
-        if not hasattr(obj, backend_name):
-            raise RuntimeError(f'Attempting to load the storages to the {backend_name.upper()} device '
-                               f'but torch.storage._StorageBase.{backend_name}() or '
-                               f'torch.storage.TypedStorage.{backend_name}() is not generated. '
-                               'Please use torch.utils.generate_methods_for_privateuse1_backend '
-                               f'to generate storage.{backend_name}() method first.')
-        device_index = _validate_privateuse1_device(location, backend_name)
-        return getattr(obj, backend_name)(device_index)
+def _deserialize(obj, location):
+    for backend_name in ['cuda', 'hpu', 'xpu', torch._C._get_privateuse1_backend_name()]:
```

Comment:
We should not limit device type for `_deserialize`. It is better to accept a parameter `device_type` in `_deserialize` to make the code more generic and can be shared across other backends. So suggest to change to `deserialize(device, obj, location)`

Reply:
done